go/libraries/doltcore/sqle/history_table.go

```diff
@@ -596,6 +600,9 @@ func rowConverter(srcSchema, targetSchema sql.Schema, h hash.Hash, meta *datas.C
 			// TODO: we could do a projection to convert between types in some cases
 			if srcSchema[srcIdx].Type.Equals(targetSchema[i].Type) {
 				srcToTarget[srcIdx] = i
+			} else {
+				ctx.Warn(1246, "Unable to convert field %s in historical rows because its type (%s) doesn't match "+
```

Comment:
Probably want to limit this warning to once per table / statement?

Reply:
Good call – I change this to only emit one warning per affected column.